manifests/helm-profiles/openshift-ambient.yaml

```diff
@@ -8,17 +8,14 @@ global:
   seLinuxOptions:
     type: spc_t
 cni:
+  enabled: true
   ambient:
     enabled: true
+  logLevel: info
+  provider: "multus"
   cniBinDir: /var/lib/cni/bin
   cniConfDir: /etc/cni/multus/net.d
   chained: false
   cniConfFileName: "istio-cni.conf"
-  logLevel: info
-  provider: "multus"
-pilot:
```

Comment:
I don't think this should move. Technically i think it will work its just a bit confusing.

The problem is we currently share the same profile between all charts, which is a _bit_ wonky but technically works.

We then allow flattening `pilot` and `cni`.

These values are meant to be passed to the pilot chart, they just (somewhat confusingly) are passed as pilot.cni. So because we unwrap that, we happen to get `cni` which happens to conflict with the CNI chart.

A more clean approach would be to have a profile-per-chart, but its a pain to maintain.

Reply:
> I don't think this should move. Technically i think it will work its just a bit confusing.
> 
> The problem is we currently share the same profile between all charts, which is a _bit_ wonky but technically works.
> 
> We then allow flattening `pilot` and `cni`.
> 
> These values are meant to be passed to the pilot chart, they just (somewhat confusingly) are passed as pilot.cni. So because we unwrap that, we happen to get `cni` which happens to conflict with the CNI chart.
> 
> A more clean approach would be to have a profile-per-chart, but its a pain to maintain.

In the case of the openshift profile, I think there is no override of any pilot value.That is why I deleted those in this case. Also, checking in the repo I only find the use of `pilot.cni*` in some testdata yaml but instead `.Values.cni.enabled` and `.Values.cni.provider` are used across all charts now.

Reply:
Yeah - these are pilot values, and used nowhere else. They should be scoped exclusively to `pilot`.

1. `helm install pilot --set cni.enabled`
2. `helm install <thing-that-composes-pilot-by-name> --set pilot.cni.enabled`

should both work in normative helm. but since we have/are slowly undoing our own scoping conventions it's not that simple. 

As I mentioned in 
https://github.com/istio/istio/pull/52036#discussion_r1676562286
if `pilot.cni` is uniquely a problem because we are eagerly unwrapping `pilot`, I think we can just manually rescope/munge `pilot.cni` first, then unwrap what's left of `pilot` as we do currently.

Reply:
@bleggett  Should I do this replacement of `cni.provider` in this PR or do it in another PR to not mix the things? I agree that if the only value used in all the charts is multus and is only used to set openshift specific configuration does not make sense to have it there and should set those values by using `values.platform=openshift`. Regarding `enable` will be only key remaining under pilot, can check as you metion to `cniEnabled` a leave it under pilot?

Reply:
If the variable move is left out of this PR, and those things are done in a follow-up, that's fine with me.

Reply:
I'll add it - can you remove https://github.com/istio/istio/blob/49abfa448f43cfd61ed85ba00b640cdd3a47befd/manifests/helm-profiles/openshift.yaml and the resulting value moves, deferring those to followups? 

That should reduce the delta here.

Reply:
sure, ill try to do that this weekend if not Monday morning will be done

Reply:
> sure, ill try to do that this weekend if not Monday morning will be done

Don't do it over the weekend - if you do I'll be tempted to look at it :P

Reply:
OK, changes are done. To summarise the changes on this PR and what is pending for the next one:
Fix:

- `.Values.cni` errors in helm charts
- Fix incorrect replace in `apiVersion: k8s.cni.cncf.io/v1`
- Fix `.Values.pilot.cni.enabled` and `.Values.pilot.cni.provider` incorrect replace that affect login in helm charts


Pending for the next PR:
- Replace the use of  `cni.provider` because the only use is to set values applicable to openshift. Replace by the use of `platform=openshift`
- Replace `pilot.cni.enabled` for `pilot. cniEnabled` to avoid warp cni inside pilot
- I'll check if there is another helm value that can be simplified and check in the docs to see if we need to update some of the docs also